tests/test_images.py

```diff
@@ -26,52 +31,60 @@
     remote = ("https://github.com/aio-libs/aiodocker/"
               "raw/master/tests/docker/docker_context.tar")
 
-    tag = "image_from_tar:1.0"
+    tag = "{}:1.0".format(_random_name())
     params = {'tag': tag, 'remote': remote, 'stream': True}
     stream = await docker.images.build(**params)
 
-    async for output in stream:
-        if "Successfully tagged image_from_tar:1.0\n" in output:
-            break
+    async for _ in stream:  # noqa: F841
+        pass
 
     image = await docker.images.get(tag)
     assert image
 
 
 @pytest.mark.asyncio
 async def test_history(docker):
-    name = "image:1.0"
+    name = "busybox:latest"
+    await docker.images.pull(from_image=name)
     history = await docker.images.history(name=name)
     assert history
 
 
 @pytest.mark.asyncio
 async def test_list_images(docker):
-    images = await docker.images.list()
-    assert images
+    name = "busybox:latest"
+    await docker.images.pull(from_image=name)
+    images = await docker.images.list(filter=name)
+    assert len(images) == 1
 
 
 @pytest.mark.asyncio
 async def test_tag_image(docker):
-    repository = "registry:5000/image"
-    name = "image:1.0"
+    name = "busybox:latest"
+    repository = _random_name()
+    await docker.images.pull(from_image=name)
     await docker.images.tag(name=name, repo=repository, tag="1.0")
     await docker.images.tag(name=name, repo=repository, tag="2.0")
     image = await docker.images.get(name)
-    assert len(image['RepoTags']) == 3
+    assert len([x for x in image['RepoTags'] if x.startswith(repository)]) == 2
 
 
 @pytest.mark.asyncio
 async def test_push_image(docker):
-    name = "registry:5000/image"
-    await docker.images.push(name=name)
+    name = "busybox:latest"
+    await docker.images.pull(from_image=name)
+    repository = "localhost:5000/image"
+    await docker.images.tag(name=name, repo=repository)
+    await docker.images.push(name=repository)
 
 
 @pytest.mark.asyncio
 async def test_delete_image(docker):
+    name = "busybox:latest"
+    await docker.images.pull(from_image=name)
```

Comment:
@barrachri because: test isolation. It's not a test suite that needs to be ran in a certain order anymore, they can be run alone! In any order! And to understand a single test, you don't need to read the others anymore! That's all the beauty 🌻